pinot-core/src/test/java/org/apache/pinot/queries/NullEnabledQueriesTest.java

```diff
@@ -156,131 +179,186 @@ private void setUp(TableConfig tableConfig, DataType dataType)
+        }
+      }
+    }
+    {
+      // This test case was added to validate path-code for distinct w/o order by.
+      int limit = 40;
+      String query = String.format("SELECT DISTINCT %s FROM testTable LIMIT %d", COLUMN_NAME, limit);
+      BrokerResponseNative brokerResponse = getBrokerResponse(query, queryOptions);
+      ResultTable resultTable = brokerResponse.getResultTable();
+      DataSchema dataSchema = resultTable.getDataSchema();
+      assertEquals(dataSchema,
+          new DataSchema(new String[]{COLUMN_NAME}, new ColumnDataType[]{dataType}));
+      List<Object[]> rows = resultTable.getRows();
+      // Boolean there's only true or false.
+      if (dataType != ColumnDataType.BOOLEAN) {
+        assertEquals(rows.size(), limit);
+      }
+    }
+    {
+      String query = String.format(
+          "SELECT COUNT(*) AS count, %s FROM testTable GROUP BY %s ORDER BY %s DESC LIMIT 1000", COLUMN_NAME,
+          COLUMN_NAME, COLUMN_NAME);
+      BrokerResponseNative brokerResponse = getBrokerResponse(query, queryOptions);
+      ResultTable resultTable = brokerResponse.getResultTable();
+      DataSchema dataSchema = resultTable.getDataSchema();
+      assertEquals(dataSchema, new DataSchema(new String[]{"count", COLUMN_NAME},
+          new ColumnDataType[]{ColumnDataType.LONG, dataType}));
+      List<Object[]> rows = resultTable.getRows();
 
-  @Test(priority = 7)
-  public void testQueriesWithNoDictDoubleColumnNoNullValues()
-          throws Exception {
-    ColumnDataType columnDataType = ColumnDataType.DOUBLE;
-    double baseValue = RANDOM.nextDouble();
-    boolean generateNulls = false;
-    createRecords(baseValue, generateNulls);
-    List<String> noDictionaryColumns = new ArrayList<String>();
-    noDictionaryColumns.add(COLUMN_NAME);
-    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE)
-            .setTableName(RAW_TABLE_NAME)
-            .setNoDictionaryColumns(noDictionaryColumns)
-            .build();
-    setUp(tableConfig, columnDataType.toDataType());
-    testQueries(baseValue, columnDataType, generateNulls);
+      if (dataType != ColumnDataType.BOOLEAN) {
+        int rowsCount = nullValuesExist ? 501 : 500;
+        assertEquals(rows.size(), rowsCount);
+        int i = 0;
+        for (int index = 0; index < 500; index++) {
+          Object[] row = rows.get(index);
+          assertEquals(row.length, 2);
+          if ((NUM_RECORDS - i - 1) % 2 == 1) {
+            // Null values are inserted at: index % 2 == 1. All null values are grouped into a single null.
+            i++;
+          }
+          assertEquals(row[0], 4L);
+          i++;
+        }
+      }
+      // The default null ordering is 'NULLS LAST'.
+      if (nullValuesExist) {
+        Object[] row = rows.get(rows.size() - 1);
+        assertEquals(row[0], 2000L);
+        assertNull(row[1]);
+      }
+    }
+    // TODO fix String null value handling
+    if (dataType != ColumnDataType.STRING) {
```

Comment:
looks like this query doesn't work with STRING type. might be we can take another look @nizarhejazi

Reply:
@walterddr There is a reason this query does not work w/ String, right? We cannot pass String column to aggregation functions (Count, Min, Max, Avg, etc.) and we cannot use inequality operators w/ Strings.

Reply:
Oh I see. So we are not using comparison for min/max and null checker for count. But actually fallback to the numeric handling. Is this correct?